src/battle_script_commands.c

```diff
@@ -4167,7 +4190,11 @@ static void Cmd_unknown_24(void)
     if (gBattleControllerExecFlags)
         return;
 
-    if (NoAliveMonsForPlayer())
+    #ifdef B_MULTI_BATTLE_WHITEOUT >= GEN_4
+    if (gBattleTypeFlags & (BATTLE_TYPE_MULTI | BATTLE_TYPE_INGAME_PARTNER) && NoAliveMonsForPlayerAndPartner())
+        gBattleOutcome |= B_OUTCOME_LOST;
+    #endif
+    if (!(gBattleTypeFlags & (BATTLE_TYPE_MULTI | BATTLE_TYPE_INGAME_PARTNER)) && NoAliveMonsForPlayer())
         gBattleOutcome |= B_OUTCOME_LOST;
```

Comment:
As it is right now, if `B_MULTI_BATTLE_WHITEOUT = GEN_3`, then it will not get a battle outcome if it's a multi battle and the player is out of mons, likely softlocking the game.
```suggestion
#ifdef B_MULTI_BATTLE_WHITEOUT >= GEN_4
    if (gBattleTypeFlags & (BATTLE_TYPE_MULTI | BATTLE_TYPE_INGAME_PARTNER))
    {
        if (NoAliveMonsForPlayerAndPartner())
            gBattleOutcome |= B_OUTCOME_LOST;
    }
    else
    {
        if (NoAliveMonsForPlayer())
            gBattleOutcome |= B_OUTCOME_LOST;
    }
#else
    if (NoAliveMonsForPlayer())
        gBattleOutcome |= B_OUTCOME_LOST;
#endif
```

Reply:
Probably would be better to just move `if (!(gBattleTypeFlags & (BATTLE_TYPE_MULTI | BATTLE_TYPE_INGAME_PARTNER)) && NoAliveMonsForPlayer())` inside the `#ifdef B_MULTI_BATTLE_WHITEOUT >= GEN_4` and change the `#endif` with a `#else` followed by the `if (NoAliveMonsForPlayer())` that is present by default.

I'll do it in 30 minutes.

Reply:
I recommended that way I did so there would only be 2 checks instead of 3. It's slightly less readable to make it slightly more efficient. I want to know what others think about it.

Reply:
Aren't you doing 3 checks here too though?

First you're checking if the current battle is a multi battle which then enters a second check to see if the Player and their Partner don't have any Pokémon left. Then there's the check, in the case that the current battle is not a multi battle, to see if the Player doesn't have Pokémon left.

Is that more efficient code-wise? 🤔